lib/datadog/core/ext/environment.rb

```diff
@@ -0,0 +1,25 @@
+require 'ddtrace/version'
+
+module Datadog
+  module Core
+    module Ext
+      module Environment
+        # Identity
+        LANG = 'ruby'.freeze
+        LANG_ENGINE = RUBY_ENGINE
+        LANG_INTERPRETER = "#{RUBY_ENGINE}-#{RUBY_PLATFORM}".freeze
+        LANG_PLATFORM = RUBY_PLATFORM
+        LANG_VERSION = RUBY_VERSION
+        RUBY_ENGINE = ::RUBY_ENGINE # e.g. 'ruby', 'jruby', 'truffleruby'
+        TRACER_VERSION = Datadog::VERSION::STRING
+
+        FALLBACK_SERVICE_NAME =
+          begin
+            File.basename($PROGRAM_NAME, '.*')
+          rescue StandardError
+            'ruby'
+          end.freeze
+      end
+    end
+  end
+end
```

Comment:
I'm curious, what's the rule to split between `core/environment` and `core/ext`? Is this distinction still useful?

Reply:
When I was moving this, I was wondering about this `ext` pattern for constants, specifically having them all under a global `ext/*` directory.

I'm not sure having them in a global directory is a good pattern, but I'd like to hear @delner's input before making changes to this structure.

Reply:
I'm giving this a try and moving `core/ext/environment.rb` to `core/environment/ext.rb`. I think it makes sense to keep it close to its namespace.